lib/issuer.js

```diff
@@ -258,29 +258,35 @@ function getCredentialSigningKey(credential, callback) {
       var identity = results.getIssuer;
       var keyId = null;
 
-      if(identity.sysPreferences && identity.sysPreferences.credentialSigningKey) {
+      if(identity.sysPreferences &&
+        identity.sysPreferences.credentialSigningKey) {
         keyId = identity.sysPreferences.credentialSigningKey;
       }
       if(!keyId) {
-        return callback();
+        return callback(new BedrockError(
+          'Preferred credential signing key not designated.', 'NotFound',
+          {httpStatusCode: 404, public: true}));
       }
       brKey.getPublicKey({id: keyId}, identity,
         function(err, publicKey, meta, privateKey) {
+        if(err) {
+          return callback(new BedrockError(
+            'Preferred credential signing key not found.', 'NotFound',
+            {httpStatusCode: 404, public: true}, err));
```

Comment:
This obscures other errors -- this could just be a database error or similar, not necessarily a not found error. We should check to see if it's a not found error (and we may not even need to special case it if it's handled lower down).